test_scripts/Defects/5_0/2445_SDL_doesn_t_resume_status_of_subscribtion_on_wayPoint_related_data.lua

```diff
@@ -0,0 +1,133 @@
+                subAdministrativeArea = "a",
+                locality = "a",
+                subLocality = "a",
+                thoroughfare = "a",
+                subThoroughfare = "a"      
+            }
+        }  
+    }
+}
+
+-- [[ Local Functions ]]
+local function cleanSessions()
+    for i = 1, common.getAppsCount() do
+      test.mobileSession[i] = nil
+    end
+    utils.wait()
+end
+
+local function pTUpdateFunc(tbl)
+    local OWgroup = {
+        rpcs = {
+            GetWayPoints = {
+                hmi_levels = { "BACKGROUND", "FULL", "LIMITED" }
+            },
+            SubscribeWayPoints = {
+                hmi_levels = { "BACKGROUND", "FULL", "LIMITED" }
+            },
+            UnsubscribeWayPoints = {
+                hmi_levels = { "BACKGROUND", "FULL", "LIMITED" }
+            },
+            OnWayPointChange =  {
+                hmi_levels = { "BACKGROUND", "FULL", "LIMITED" }
+            }
+        }
+    }
+    tbl.policy_table.functional_groupings["NewTestCaseGroup"] = OWgroup
+    tbl.policy_table.app_policies[config.application1.registerAppInterfaceParams.fullAppID].groups = { "Base-4", "NewTestCaseGroup" }
+end
+
+local function subscribeWayPoints()
+    local cid = common.getMobileSession():SendRPC("SubscribeWayPoints", {})
+    common.getHMIConnection():ExpectRequest("Navigation.SubscribeWayPoints")
+    :Do(function(_, data)
+        common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS",{})
+    end)
+    common.getMobileSession():ExpectResponse(cid, { success = true , resultCode = "SUCCESS" })
+    common.getMobileSession():ExpectNotification("OnHashChange")
+end
+
+local function onWayPointChange()
+    common.getHMIConnection():SendNotification("Navigation.OnWayPointChange", notifParams)
+    common.getMobileSession():ExpectNotification("OnWayPointChange", notifParams)
+end
+
+local function closeMobileSession()
+    local cid = common.getMobileSession():SendRPC("UnregisterAppInterface", {})
+    common.getHMIConnection():ExpectRequest("Navigation.UnsubscribeWayPoints")
+    :Do(function(_, data)
+        common.getMobileSession():ExpectResponse(cid, { success = true , resultCode = "SUCCESS" })
+        :Do(function(_, data)
+          common.getMobileSession():Stop()
+        end)
+    end)
+end
+
+--[[ Scenario ]]
+runner.Title("Preconditions")
+runner.Step("Clean environment", common.preconditions)
+runner.Step("Start SDL, HMI, connect Mobile, start Session", common.start)
+runner.Step("Register App", common.registerApp, { 1 })
+runner.Step("PTU", common.policyTableUpdate, { pTUpdateFunc })
+runner.Step("Activate App", common.activateApp, { 1 })
+runner.Step("SubscribeWayPoints", subscribeWayPoints)
+runner.Step("On Way Point Change", onWayPointChange)
+runner.Step("Close mobile session", closeMobileSession)
+runner.Step("Clean sessions", cleanSessions)
+
+-- [[ Test ]]
+runner.Title("Test")
+runner.Step("Register App", common.registerAppWOPTU, { 1 })
```

Comment:
@TMyza Test scenario does not check resumption of HMI level of unexpected disconnected mobile application and does not check restoring of its subscription to onWayPointChange HMI event.
Script is incorrect.